src/aeso_fcode_to_fate.erl

```diff
@@ -1605,7 +1651,23 @@ bb(_Name, Code) ->
     Blocks  = lists:flatmap(fun split_calls/1, Blocks1),
     Labels  = maps:from_list([ {Ref, I} || {I, {Ref, _}} <- with_ixs(Blocks) ]),
     BBs     = [ set_labels(Labels, B) || B <- Blocks ],
-    maps:from_list(BBs).
+    maps:from_list(dbg_loc_filter(BBs)).
+
+%% Filter DBG_LOC instructions to keep one instruction per line
+dbg_loc_filter(BBs) ->
+    dbg_loc_filter(BBs, [], [], sets:new()).
+
+dbg_loc_filter([], _, AllBlocks, _) ->
+    lists:reverse(AllBlocks);
+dbg_loc_filter([{I, []} | Rest], AllOps, AllBlocks, DbgLocs) ->
+    dbg_loc_filter(Rest, [], [{I, lists:reverse(AllOps)} | AllBlocks], DbgLocs);
+dbg_loc_filter([{I, [Op = {'DBG_LOC', _, _} | Ops]} | Rest], AllOps, AllBlocks, DbgLocs) ->
+    case sets:is_element(Op, DbgLocs) of
```

Comment:
Why not just check if only the last instruction is of the same loc?

Reply:
At this point I would assume DBG_LOCs are added correctly, and we only want to get rid of redundancies.

Reply:
>Why not just check if only the last instruction is of the same loc?

Do you mean checking for each `DBG_LOC` instruction at position `I` for the instruction at position `I - 1`?

If that's what mean, then it would only work for the following case:

```
DBG_LOC N
DBG_LOC N
instrucions
```

When multiple expressions are on the same line, the generated instructions would look like this:

```
DBG_LOC N
instructions
DBG_LOC N
instructions
```

This is why I had to use a `set`.

Reply:
That's why I wanted columns, but fair for now